dev/io.openliberty.io.smallrye.config/src/io/openliberty/microprofile/config/internal/extension/OLSmallRyeConfigExtension.java

```diff
@@ -10,15 +10,59 @@
  *******************************************************************************/
 package io.openliberty.microprofile.config.internal.extension;
 
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
 import javax.enterprise.inject.spi.Extension;
 
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.ConfigurationPolicy;
 
 import com.ibm.ws.cdi.extension.WebSphereCDIExtension;
 
+import io.openliberty.checkpoint.spi.CheckpointPhase;
 import io.smallrye.config.inject.ConfigExtension;
 
 @Component(service = WebSphereCDIExtension.class, configurationPolicy = ConfigurationPolicy.IGNORE, property = { "service.vendor=IBM" }, immediate = true)
 public class OLSmallRyeConfigExtension extends ConfigExtension implements Extension, WebSphereCDIExtension {
+    static final CheckpointPhase phase = CheckpointPhase.getPhase();
+    static final ThreadLocal<AtomicBoolean> recordingStopped = new ThreadLocal<AtomicBoolean>() {
+        @Override
+        protected AtomicBoolean initialValue() {
+            return new AtomicBoolean();
+        }
+    };
+
+    @Override
+    protected void validate(@Observes AfterDeploymentValidation adv) {
+        boolean stopped = stopRecordingReads();
+        try {
+            super.validate(adv);
+        } finally {
+            startRecordingReads(stopped);
+        }
+    }
+
+    static void startRecordingReads(boolean stopped) {
```

Comment:
these method names are confusing ... a method called "start" that takes a parameter called "stopped" which actually sets a thread local called "recordingStopped" to false?

Reply:
An alternative would be to have `stopRecordingReads` to return an `AutoCloseable` and the `close()` method only resumes the reads if `stopRecordingReads` actually stopped them.  Then it would be code like this:

```java
    try (AutoCloseable stoppedRecording = stopRecordingReads()){
        // do stuff while not recording reads
    }
```

Reply:
I think this would be better. Maybe `pauseRecordingReads()` or `withReadRecordingPaused()` if the implication is that recording will always be resumed.

Reply:
Renamed method. Thanks!